File: Health-app-main/Health-app-main/frontend/src/types/appointment.ts
export enum AppointmentStatus {
  PENDING = 'PENDING',
  SCHEDULED = 'SCHEDULED',
  CONFIRMED = 'CONFIRMED',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED',
  NO_SHOW = 'NO_SHOW'
}

export enum AppointmentType {
  IN_PERSON = 'IN_PERSON',
  VIRTUAL = 'VIRTUAL'
}

// Frontend appointment format used in UI
export interface Appointment {
  id: string;
  title: string;
  startTime: string;
  endTime: string;
  status: AppointmentStatus;
  type: AppointmentType;
  doctor: {
    id: string;
    name: string;
    specialty: string;
  };
  patientId?: string;
  doctor_id?: string; // For backward compatibility
  patient_id?: string; // For backward compatibility 
  appointment_date?: string; // For backward compatibility
  appointment_type?: string; // For backward compatibility
  duration_minutes?: number; // For backward compatibility
  reason?: string;
  notes?: string;
  location?: string;
  meetingLink?: string;
  instructions?: string;
  insurance?: {
    provider: string;
    policyNumber: string;
    groupNumber?: string;
  };
  history: {
    timestamp: string;
    action: string;
    by: string;
  }[];
}

// Data needed to create a new appointment
export interface AppointmentCreate {
  title?: string;
  doctorId: string;
  patientId: string;
  startTime: string;
  endTime: string;
  type: AppointmentType;
  reason?: string;
  notes?: string;
  location?: string;
  meetingLink?: string;
  insurance?: {
    provider: string;
    policyNumber: string;
    groupNumber?: string;
  };
  // Backward compatibility fields
  doctor_id?: string;
  patient_id?: string;
  appointment_type?: AppointmentType;
  duration_minutes?: number;
}

// Fields that can be updated
export interface AppointmentUpdate {
  title?: string;
  doctorId?: string;
  startTime?: string;
  endTime?: string;
  type?: AppointmentType;
  status?: AppointmentStatus;
  reason?: string;
  notes?: string;
  location?: string;
  meetingLink?: string;
} 

File: Health-app-main/Health-app-main/backend/src/models/appointment.py
from datetime import datetime, timedelta
from enum import Enum
from typing import Optional
from pydantic import BaseModel, Field

# Appointment model for the Hospital Management System.
# This file defines the structure and behavior of appointments in the system, ensuring seamless integration with IoT devices for health monitoring.

class AppointmentStatus(str, Enum):
    SCHEDULED = "scheduled"
    CONFIRMED = "confirmed"
    CANCELLED = "cancelled"
    COMPLETED = "completed"
    NO_SHOW = "no_show"

class AppointmentType(str, Enum):
    REGULAR = "regular"
    FOLLOW_UP = "follow_up"
    URGENT = "urgent"
    CONSULTATION = "consultation"

class AppointmentBase(BaseModel):
    patient_id: str
    doctor_id: str
    appointment_date: datetime
    appointment_type: AppointmentType
    reason: str
    duration_minutes: int = Field(default=30, ge=15, le=120)
    notes: Optional[str] = None

class AppointmentCreate(AppointmentBase):
    pass

class AppointmentUpdate(BaseModel):
    appointment_date: Optional[datetime] = None
    appointment_type: Optional[AppointmentType] = None
    reason: Optional[str] = None
    duration_minutes: Optional[int] = Field(default=None, ge=15, le=120)
    notes: Optional[str] = None
    status: Optional[AppointmentStatus] = None

# Appointment class that represents a scheduled appointment between a patient and a doctor.
# It includes methods for creating, updating, and retrieving appointment details, supporting real-time health data synchronization.

class Appointment(AppointmentBase):
    id: str
    status: AppointmentStatus = AppointmentStatus.SCHEDULED
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True 